lib/openai/http.rb

```diff
@@ -86,13 +86,24 @@ def conn(multipart: false)
 
     def uri(path:)
       if azure?
-        base = File.join(@uri_base, path)
-        "#{base}?api-version=#{@api_version}"
+        azure_uri(path)
       else
         File.join(@uri_base, @api_version, path)
       end
     end
 
+    def azure_uri(path)
+      base = File.join(@uri_base, path)
+
+      # Remove the deployment to support assistants for azure
+      if path.include?("/assistants") || path.include?("/threads")
+        base = base.gsub(%r{/deployments/.+/},
```

Comment:
I believe this should be:
```suggestion
        base = base.gsub(%r{/deployments/[^\/]+/},
```
It should find anything that's not a `/`, otherwise it'll replace any subdirectory but the last.

For example:

```
$ '/openai/deployments/gpt-4-turbo-0125/assistants/asst_Vw5SZ5WPQJYGGDZD7wbrN78R'.gsub(%r{/deployments/.+/}, '/')
=> "/openai/asst_Vw5SZ5WPQJYGGDZD7wbrN78R"
```

I caught this when using your change and trying to delete an assistant.

Reply:
Good catch!